modules/generic/src/main/scala/pureconfig/generic/semiauto.scala

```diff
@@ -13,4 +13,32 @@ object semiauto {
 
   final def deriveConvert[A](implicit reader: Lazy[DerivedConfigReader[A]], writer: Lazy[DerivedConfigWriter[A]]): ConfigConvert[A] =
     ConfigConvert.fromReaderAndWriter(Derivation.Successful(reader.value), Derivation.Successful(writer.value))
+
+  final def deriveEnumerationReader[A](
+    implicit
+    readerBuilder: Lazy[EnumerationConfigReaderBuilder[A]]): ConfigReader[A] =
+    deriveEnumerationReader(_.toLowerCase)
+  final def deriveEnumerationWriter[A](
+    implicit
+    writerBuilder: Lazy[EnumerationConfigWriterBuilder[A]]): ConfigWriter[A] =
+    deriveEnumerationWriter(_.toLowerCase)
+  final def deriveEnumerationConvert[A](
+    implicit
+    readerBuilder: Lazy[EnumerationConfigReaderBuilder[A]],
+    writerBuilder: Lazy[EnumerationConfigWriterBuilder[A]]): ConfigConvert[A] =
+    deriveEnumerationConvert(_.toLowerCase)
+
+  final def deriveEnumerationReader[A](transformName: String => String)(
```

Comment:
I'm not very fond of polluting the semiauto API with variable-argument alternatives. The approach we took with the `pureconfig` package object was already shown many times not to be scalable :/

Hints win here on this specific case in terms of flexibility. With them you can customize an "unlimited" number of aspects while providing only a single instance as an implicit instance. Adding "hint" type classes for enumerations may be a bit overkill and may also be optimizing prematurely, though. Upgrading this to use a  magnet pattern here may be a reasonable solution.

Can you see any alternative to having an overloaded method here? Otherwise we might be ok as it is for now...

Reply:
It's be a cheat but we could make the to lower the default argument to this version and get rid of the nocarg version.

Reply:
I think both the magnet pattern and the default argument are good alternatives to avoid the overloading. I went this way to be able to call `deriveEnumerationReader[T]` without the extra `()`, for consistency with the `deriveReader[T]` method calls. We won't be able to do it with default arguments, and I also think that using the magnet pattern would force us to have the extra `()` for the default case (not sure about this).

I think it's premature to add an extra hint for enumerations.